### PR TITLE
Fixes #26878 - commit deprecations

### PR DESCRIPTION
**Pull request by jeremylenz**
This PR completes actual removal of features associated with the following deprecation warnings:

```
[vagrant@centos7-katello-devel tool_belt]$ ./tools.rb check-deprecation-warnings configs/katello/3.14.yaml
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.7-compliant syntax, but you are running 2.5.5.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.

Scanning for deprecation warnings in /home/vagrant/tool_belt/repos/katello/3.14.1/katello/app/**/*.rb, /home/vagrant/tool_belt/repos/katello/3.14.1/katello/lib/**/*.rb

Deprecation warnings were found with versions that are outdated, please check below and ensure both the deprecated functionality and the deprecation warning are removed if the deprecation is outdated in this release

Deprecation warning at /home/vagrant/tool_belt/repos/katello/3.14.1/katello/app/controllers/katello/api/v2/host_subscriptions_controller.rb line 161 is marked for removal in version 3.5, which is less than or equal to the specified version of 3.14.1.

Deprecation warning at /home/vagrant/tool_belt/repos/katello/3.14.1/katello/app/controllers/katello/api/v2/content_view_versions_controller.rb line 291 is marked for removal in version 3.4, which is less than or equal to the specified version of 3.14.1.

Deprecation warning at /home/vagrant/tool_belt/repos/katello/3.14.1/katello/app/controllers/katello/api/v2/repositories_controller.rb line 409 is marked for removal in version 3.3, which is less than or equal to the specified version of 3.14.1.

Deprecation warning at /home/vagrant/tool_belt/repos/katello/3.14.1/katello/app/controllers/katello/api/v2/subscriptions_controller.rb line 204 is marked for removal in version 2.6, which is less than or equal to the specified version of 3.14.1.
```

Does not address the following future deprecation:

```
Deprecation warnings were found marked for removal for the *next* release version 3.15. Please check below and file issues aligned to 3.15 to remove the functionality and the deprecation warning. This will ensure the functionality is removed in time for the next release 3.15. The deprecation warning should stay in this release, only file issues for its removal at this time.

Deprecation warning at /home/vagrant/tool_belt/repos/katello/3.14.1/katello/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb line 70 is marked for removal in version 3.15, which is the *next* release version.
```

### CODE REVIEW

**Comment by jeremylenz** [comment]
Test failures seem related; working on it

**Comment by mmoll** [comment]
With GH-8486 the following are also visible:

```
Your API call uses deprecated behavior, Export is being deprecated and will be removed in a future version of Katello. Use hammer content-view version export instead. (called from export at app/controllers/katello/api/v2/content_view_versions_controller.rb:91)
Your API call uses deprecated behavior, The parameter ignore_global_proxy will be removed in a future Katello release. Please update to use the http_proxy_policy parameter. (called from check_ignore_global_proxy at app/controllers/katello/api/v2/repositories_controller.rb:481)
```

**Comment by jjeffers** [comment]
> With [GH-8486](https://github.com/Katello/katello/pull/8486) the following are also visible:

> ```
> Your API call uses deprecated behavior, The parameter ignore_global_proxy will be removed in a future Katello release. Please update to use the http_proxy_policy parameter. (called from check_ignore_global_proxy at app/controllers/katello/api/v2/repositories_controller.rb:481)
> ```
I think that one will be changed to be removed in 2.0. per  https://projects.theforeman.org/issues/29205

File: app/controllers/katello/api/v2/content_view_versions_controller.rb [hunk 1]
```diff
@@ -52,7 +52,6 @@ def show
     api :POST, "/content_view_versions/:id/promote", N_("Promote a content view version")
     param :id, :number, :desc => N_("Content view version identifier"), :required => true
     param :force, :bool, :desc => N_("force content view promotion and bypass lifecycle environment restriction")
-    param :environment_id, :number, :deprecated => true, :desc => N_("LifeCycle Environment identifier")
```
Comment (jjeffers) [inline app/controllers/katello/api/v2/content_view_versions_controller.rb]: This seems to cause a problem for the hammer katello client.

I get a 400 Bad Request, 

```
[ INFO 2020-03-09T15:55:59 API] POST /katello/api/content_view_versions/2/promote
[DEBUG 2020-03-09T15:55:59 API] Params: {
    "environment_id" => "3",
             "force" => nil
}
[DEBUG 2020-03-09T15:55:59 API] Headers: {}
[DEBUG 2020-03-09T15:55:59 API] Using authenticator: HammerCLIForeman::Api::InteractiveBasicAuth
[ERROR 2020-03-09T15:55:59 API] 400 Bad Request
[DEBUG 2020-03-09T15:55:59 API] {
    "displayMessage" => "Could not find environments for promotion",
            "errors" => [
        [0] "Could not find environments for promotion"
    ]
}
```

Comment (jjeffers) [reply app/controllers/katello/api/v2/content_view_versions_controller.rb]: I'm not sure this should block this PR, but perhaps we need to make sure the hammer client is updated once this PR is merged? Thoughts @chris1984 @akofink

**Comment by mmoll** [comment]
It would be nice to have a version instead of "future". ;)

**Comment by jjeffers** [comment]
> It would be nice to have a version instead of "future". ;)

@mmoll  Should we go ahead and add those versions to the warnings now? Or just tackle that when we schedule https://projects.theforeman.org/issues/29205?

File: app/controllers/katello/api/v2/content_view_versions_controller.rb (hunk 1 above)
Comment (chris1984) [reply app/controllers/katello/api/v2/content_view_versions_controller.rb]: I am fine with having an update to the hammer client after this is merged.

Comment (jjeffers) [reply app/controllers/katello/api/v2/content_view_versions_controller.rb]: I will create an issue for that update, then.

**Comment by mmoll** [comment]
@jjeffers aw, sorry.... now I understand that ticket. Can be fixed with it later, of course. :+1:

**Comment by jturel** [comment]
Was this one not picked up by the deprecation finder?

https://github.com/Katello/katello/blob/21bffbadc4cc9e0acb29213704be6ab9840b66d5/app/controllers/katello/api/v2/host_subscriptions_controller.rb#L58

**Comment by jeremylenz** [comment]
@jturel `configs/katello/3.14.yaml` was the latest version available when I ran it

**Comment by jturel** [comment]
> @jturel `configs/katello/3.14.yaml` was the latest version available when I ran it

Hmm - 3.15.yaml should be there for a while now. I thought the tool was reporting version+1 deprecations too

**Comment by jeremylenz** [comment]
@jturel It did report one for 3.15 (the output is in the first PR comment above.)

**Pull request merged**